x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -151,6 +155,33 @@ export function App({
 
   useEffect(() => {
     onAppLeave((actions) => {
+      if (
+        initialContext &&
+        'vizEditorOriginatingAppUrl' in initialContext &&
+        initialContext.vizEditorOriginatingAppUrl
+      ) {
+        const initialDocFromContextHasChanged = !isLensEqual(
+          initialDocFromContext,
+          lastKnownDoc,
+          data.query.filterManager.inject,
+          datasourceMap
+        );
+        if (!initialDocFromContextHasChanged) {
+          return actions.default();
+        } else {
+          return actions.confirm(
+            i18n.translate('xpack.lens.app.goBackModal.message', {
+              defaultMessage:
+                'The changes you have made here are not backwards compatible with your original {contextOriginatingApp}. Are you sure you want to discard these unsaved changes and return to {contextOriginatingApp}?',
+              values: { contextOriginatingApp },
+            }),
+            i18n.translate('xpack.lens.app.goBackModal.title', {
+              defaultMessage: 'Discard changes?',
+            })
```

Comment:
@MichaelMarcialis thank you for bringing this up because I wanted to discuss it with you and I forgot it!
The reason that I went with the blue one is because this is how we are handling these cases. Every time that a user tries to exit the lens editor and has unsaved values we display a modal with a blue button. Why is this case different?

Reply:
In the case of a user attempting to go back to TSVB after making changes to the visualization in Lens, I wanted to be sure they understood that this would be a destructive action (as the changes they've made will be lost in the process). I felt the red colored button did a good job of conveying that.

That said, you raise a good point about the situation where users attempt to leave Lens without saving. Currently Lens (also Maps, Dashboard) uses a blue colored "Confirm" button. Personally, I would consider this a destructive action as well. Off the top of my head, I can imagine two possible ways to change the unsaved changes modal in order to create better consistency:

1. We could update the wording and button color of the unsaved changes modal to more closely match what is being suggested above for the TSVB backward compatibility modal shown above.
2. Alternatively, we could change the primary action of the unsaved changes modal to be "Save" (using a blue colored button) which would shortcut the user to the save modal flow. Then a secondary red colored empty button could be allow users to "Discard unsaved changes" and leave Lens.

Personally, I lean towards #1 for the sake of simplicity and consistency with other areas of Kibana, but curious what ya'll think. CCing @ghudgins in case he has any thoughts.

Reply:
I was thinking the same, +1 on the 1st point from me!

Reply:
agree with you both

Reply:
Done 
![image](https://user-images.githubusercontent.com/17003240/152776016-3a7a4eba-6a1e-4512-af8e-cba88e0b03fa.png)